### PR TITLE
Minor cleanup as the years go by (docker-compose version, etc)

### PR DESCRIPTION
**Pull request by rfay**
# The Problem/Issue/Bug

I just took a quick pass through and noted  a few things like needing to remove `version` from docker-compose

### Summary by CodeRabbit

- New Feature: Upgraded PHP version to 8.1, MariaDB version to 10.4, and Node.js version to 18 in DDEV configuration for improved performance and compatibility.
- Refactor: Optimized performance strategies and service ports in the DDEV setup for better resource management.
- Chore: Removed unnecessary version declarations from docker-compose files for cleaner codebase.
- New Feature: Added `chromedriver` service in the testing environment using `drupalci/chromedriver:production` image, enhancing testing capabilities.

### CODE REVIEW

File: .ddev/config.yaml [hunk 1]
```diff
@@ -3,22 +3,12 @@ type: drupal9
 docroot: web
 php_version: "8.1"
 webserver_type: nginx-fpm
-router_http_port: "80"
-router_https_port: "443"
-xdebug_enabled: false
-additional_hostnames: []
-additional_fqdns: []
```
Comment (rfay) [inline .ddev/config.yaml:10]: These just aren't useful to have in here

File: .ddev/config.yaml [hunk 2]
```diff
@@ -3,22 +3,12 @@ type: drupal9
...
 database:
   type: mariadb
-  version: "10.3"
```
Comment (rfay) [inline .ddev/config.yaml:13]: Better to use current required version? Doesn't d10 require 10.4? It's now the default anyway.

File: .ddev/config.yaml [hunk 3]
```diff
@@ -3,22 +3,12 @@ type: drupal9
...
-nfs_mount_enabled: false
-mutagen_enabled: false
-use_dns_when_possible: true
-composer_version: "2"
+  version: "10.4"
 web_environment: []
-nodejs_version: "16"
```
Comment (rfay) [inline .ddev/config.yaml:19]: 18 is now the default

File: .ddev/docker-compose.ports.yaml [hunk 4]
```diff
@@ -1,5 +1,4 @@
 # Expose port 3000 and 5000 of DDEV's web container.
-version: '3.6'
```
Comment (rfay) [inline .ddev/docker-compose.ports.yaml:2]: `version` is obsolete

File: .ddev/config.yaml [hunk 5]
```diff
@@ -3,22 +3,12 @@ type: drupal9
...
-router_http_port: "80"
-router_https_port: "443"
-xdebug_enabled: false
-additional_hostnames: []
-additional_fqdns: []
 database:
   type: mariadb
-  version: "10.3"
-nfs_mount_enabled: false
-mutagen_enabled: false
```
Comment (rfay) [inline .ddev/config.yaml:15]: `mutagen_enabled` is no longer valid syntax

**Comment by shaal** [comment]
Thank you @rfay
I added this update to https://github.com/shaal/DrupalPod/pull/123

**Pull request closed without merging**